website/project/tasks.py

```diff
@@ -38,6 +38,26 @@ def on_node_updated(node_id, user_id, first_save, saved_fields, request_headers=
     if node.get_identifier_value('doi') and bool(node.IDENTIFIER_UPDATE_FIELDS.intersection(saved_fields)):
         node.request_identifier_update(category='doi')
 
+@celery_app.task(ignore_results=True)
+def update_files_count(node_id):
+    AbstractNode = apps.get_model('osf.AbstractNode')
+    node = AbstractNode.load(node_id)
+
+    if node is None:
+        return
+
+    field = AbstractNode._meta.get_field('modified')
+    # Do not update modified upon save()
+    field.auto_now = False
+
+    if node.is_registration:
+        node.files_count = node.files.count()
```

Comment:
Seems like this should be done the same way as the migration, and should filter out deleted files.

Reply:
Deleted (project) files are stored on registrations.